pkg/lib/registry/registry.go

```diff
@@ -342,9 +342,45 @@ func (r RegistryUpdater) DeprecateFromRegistry(request DeprecateFromRegistryRequ
 		return fmt.Errorf("unable to migrate database: %s", err)
 	}
 
-	deprecator := sqlite.NewSQLDeprecatorForBundles(dbLoader, request.Bundles)
+	// Check if all bundlepaths are valid
+	var toDeprecate []string
+
+	dbQuerier := sqlite.NewSQLLiteQuerierFromDb(db)
+	registryBundles, err := dbQuerier.ListBundles(context.TODO())
+	if err != nil {
+		if !request.Permissive {
+			return fmt.Errorf("cannot verify bundle presence in registry: %v", err)
+		}
+		toDeprecate = request.Bundles
+	}
+
+	if registryBundles != nil {
+		validPath := make(map[string]struct{})
+		for _, b := range registryBundles {
+			validPath[b.BundlePath] = struct{}{}
+		}
+
+		missingPaths := make([]string, 0)
+		toDeprecate = make([]string, 0)
+		for _, bundlePath := range request.Bundles {
+			if _, ok := validPath[bundlePath]; !ok {
+				missingPaths = append(missingPaths, bundlePath)
+				continue
+			}
+			toDeprecate = append(toDeprecate, bundlePath)
+		}
+
+		if len(missingPaths) > 0 {
+			if !request.Permissive {
```

Comment:
IIRC, permissive should decide to return an error at the top-level command, not inform whether an error is returned. I think the proper thing to do in this case is to aggregate all errors and return them at the end of the method.